src/borg/repository.py

```diff
@@ -724,9 +724,51 @@ def __contains__(self, id):
         return id in self.index
 
     def list(self, limit=None, marker=None):
+        """
```

Comment:
Hm. A principial problem with this approach is that it re-reads all the segments again, even if the ArchiveChecker would actually not do it, ie. no --verify-data specified.

Maybe we can pass a "hint" or something (via an environment variable or similar for remotes, simple flag for local) which we want:

- list() will be traversed linearly
- list() will only be used for hashtable population